scripts/preinstall.js

```diff
@@ -17,7 +19,35 @@ const configText = `module.exports = {
   localPlugins: [],
 }`;
 
+// simple utility to remove whitespace from string
+function trim(string) {
+  return string.replace(/^\s+|\s+$/g, '');
+}
+
 try {
+  // check minimum version of npm
+  let npmVersion = execSync('npm --version', {
+    encoding: 'utf8'
+  });
+  let nodeVersion = execSync('node --version', {
```

Comment:
Kinda funny. Is this really the way to get the node version from inside a js program?!

Reply:
Good call! I went this way just because it was necessary for npm and so I was lazy. Turns out (not surprisingly) there is a `process.version`. Updated.